### PR TITLE
3.3 - Query modifiers

### LINKED ISSUE #7259: 3.x ORM modifier() does not work as expected
```modifier()``` is being ignored on these two calls:

```php
$this->query()
    ->delete()
    ->modifier('LOW_PRIORITY')
    ->where(['valid_thru <' => new Time()])
    // ->sql(): 'DELETE FROM tokens WHERE valid_thru < :c0'
    ->execute();
```
```php
$this->query()
    ->update()
    ->modifier('DELAYED')
    ->set(['valid_thru' => $validThru])
    ->where(['identifier' => $identifier])
// ->sql(): 'UPDATE tokens SET valid_thru = :c0 WHERE identifier = :c1'
    ->execute();
```
By ignored I mean... it does not trigger any error but does neither appear in the SQL statement, but should.

### PR DESCRIPTION
**Pull request by lorenzo**
This adds support for query modifiers to insert, update and delete queries. Closes #7259

### CODE REVIEW

**Comment by ionas** [comment]
Thank you!

File: src/Database/QueryCompiler.php [hunk 1]
```diff
@@ -304,20 +301,53 @@ protected function _buildValuesPart($parts, $query, $generator)
...
+     *
+     * @param array $parts The query modifier parts
+     * @param \Cake\Database\Query $query The query that is being compiled
+     * @param \Cake\Database\ValueBinder $generator the placeholder generator to be used in expressions
+     * @return string SQL fragment.
+     */
+    protected function _buildModifierPart($parts, $query, $generator)
+    {
+        if ($parts === []) {
+            return null;
```
Comment (markstory) [inline src/Database/QueryCompiler.php]: Perhaps return '' instead?

File: tests/TestCase/Database/QueryTest.php [hunk 2]
```diff
@@ -3636,6 +3636,115 @@ public function testBetweenExpressionAndTypeMap()
...
+        );
+
+        $query = new Query($this->connection);
+        $result = $query
+            ->insert(['title'])
+            ->into('articles')
+            ->values(['title' => 'foo'])
+            ->modifier(['IGNORE', 'LOW_PRIORITY']);
+        $this->assertQuotedQuery(
+            'INSERT IGNORE LOW_PRIORITY INTO <articles> \(<title>\) VALUES \(:c0\)',
```
Comment (markstory) [inline tests/TestCase/Database/QueryTest.php]: Is this going to work on postgres and sqlserver?

Comment (jhli) [inline tests/TestCase/Database/QueryTest.php]: No, I think it is specific to mariadb/mysql

**Comment by markstory** [comment]
The [Appveyor failures look relevant](https://ci.appveyor.com/project/cakephp/cakephp/build/1.0.2708)

**Comment by ionas** [comment]
Don't know if that's the reason but there is a superfluous regular whitespace after the generated ignore https://ci.appveyor.com/project/cakephp/cakephp/build/1.0.2708#L375

**Comment by lorenzo** [comment]
@markstory It seems like I rebased my branch and your commits where lost :(

I have replicated your changes myself.

**Comment by markstory** [comment]
@lorenzo That happens :smile:

**Pull request merged**